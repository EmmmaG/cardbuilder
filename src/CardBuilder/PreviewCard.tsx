import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

import { FormValue } from './BuilderForm'
import person from '../assets/person.png'

import CardRow from '../Components/CardRow'

const styles = (theme: Theme) => ({
    mainContainer: {
        width: '450px',
        display: 'inline-block',
    },
    title: {
        textAlign: 'right' as 'right',
        color: theme.palette.grey[400],
        fontSize: '18px',
        fontWeight: 500,
        marginTop: '5px',
        marginBottom: '5px',
    },
    cardContainer: {
        position: 'relative' as 'relative',
        width: '100%',
        boxShadow: `0.5px 2px ${theme.palette.grey[300]}`,
    },
    nameContainer: {
        height: '110px',
        backgroundColor: '#394F62',
    },
    name: {
        color: 'white',
        fontFamily: '"Big Caslon", "Times New Roman", "serif"',
        fontSize: '26px',
        fontWeight: 600,
        paddingTop: '55px',
        paddingLeft: '20px',
    },
    profile: {
        position: 'absolute' as 'absolute',
        zIndex: 1,
        top: '15px',
        right: '15px',
        height: '110px',
        width: '85px',
        backgroundColor: 'white',
        padding: '5px',
        border: `1px solid ${theme.palette.grey[300]}`,
        boxShadow: `1px 2px ${theme.palette.grey[300]}`,
    },
    contentsContainer: {
        padding: '20px 30px',
        backgroundColor: 'white',

    },
    postCountryContainer: {
        display: 'flex',
        justifyContent: 'space-between',
    },
})

export interface Props extends WithStyles<typeof styles> {
    contents: FormValue
    avatar?: string
}

export class PreviewCard extends React.Component<Props > {
    private renderCardContents = () => {
        const { contents, classes } = this.props
        const suburbRow = `${contents.suburb}${(contents.suburb && contents.state) ? ',' : ''} ${contents.state}`

        return (
            <div className={classes.contentsContainer}>
                <CardRow title='EMAIL' content={contents.email} />
                <CardRow title='PHONE' content={contents.phone} />
                <CardRow title='ADDRESS' content={`${contents.houseName} ${contents.street}`} />
                <CardRow content={suburbRow} />
                <div className={classes.postCountryContainer}>
                    <CardRow title='POSTCODE' content={contents.postcode} inline />
                    <CardRow title='COUNTRY' content={contents.country} inline />
                </div>
            </div>
        )
    }

    public render() {
        const { classes, contents, avatar } = this.props

        return (
            <div className={classes.mainContainer}>
                <Typography className={classes.title}>HCARD PREVIEW</Typography>
                <div className={classes.cardContainer}>
                    <div className={classes.nameContainer}>
                        <Typography className={classes.name}>{`${contents.givenName} ${contents.surname}`}</Typography>
                    </div>
                    <img src={avatar || person} className={classes.profile}/>
                    {this.renderCardContents()}
                </div>
            </div>
        )
    }
}

export default withStyles(styles)(PreviewCard)
