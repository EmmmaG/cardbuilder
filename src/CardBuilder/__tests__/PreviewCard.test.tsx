import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import PreviewCard, { Props } from '../PreviewCard'

const props: Props = {
    contents: {
        givenName: 'Jon',
        surname: '',
        email: 'Jon@somewhere.com',
        phone: '012345678',
        houseName: '',
        street: '',
        suburb: '',
        state: 'NSW',
        postcode: '1000',
        country: 'Aus',
    },
	classes: idObj,
}

describe('PreviewCard', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<PreviewCard {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
    })
})
