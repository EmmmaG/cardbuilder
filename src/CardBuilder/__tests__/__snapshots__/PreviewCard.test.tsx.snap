// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewCard renders correctly 1`] = `
<div
  className="PreviewCard-mainContainer-1"
>
  <WithStyles(Typography)
    className="PreviewCard-title-2"
  >
    HCARD PREVIEW
  </WithStyles(Typography)>
  <div
    className="PreviewCard-cardContainer-3"
  >
    <div
      className="PreviewCard-nameContainer-4"
    >
      <WithStyles(Typography)
        className="PreviewCard-name-5"
      >
        Jon 
      </WithStyles(Typography)>
    </div>
    <img
      className="PreviewCard-profile-6"
      src={
        Object {
          "process": [Function],
        }
      }
    />
    <div
      className="PreviewCard-contentsContainer-7"
    >
      <WithStyles(CardRow)
        content="Jon@somewhere.com"
        title="EMAIL"
      />
      <WithStyles(CardRow)
        content="012345678"
        title="PHONE"
      />
      <WithStyles(CardRow)
        content=" "
        title="ADDRESS"
      />
      <WithStyles(CardRow)
        content=" NSW"
      />
      <div
        className="PreviewCard-postCountryContainer-8"
      >
        <WithStyles(CardRow)
          content="1000"
          inline={true}
          title="POSTCODE"
        />
        <WithStyles(CardRow)
          content="Aus"
          inline={true}
          title="COUNTRY"
        />
      </div>
    </div>
  </div>
</div>
`;
